jmx/jmx.go

```
@@ -331,16 +331,13 @@ func Query(objectPattern string, timeoutMillis int) (result map[string]interface
 }
 
 // receiveResult checks for channels to receive result from nrjmx command.
-func receiveResult(lineCh chan []byte, queryErrors chan error, cancelFn context.CancelFunc, objectPattern string, timeout time.Duration) (result map[string]interface{}, err error) {
-	gotResult := false
+func receiveResult(lineC chan []byte, queryErrC chan error, cancelFn context.CancelFunc, objectPattern string, timeout time.Duration) (result map[string]interface{}, err error) {
 	var warn string
 	for {
 		select {
-		case line := <-lineCh:
-			gotResult = true
+		case line := <-lineC:
 			if len(line) == 0 {
 				cancelFn()
-				Close()
```

This Close call swallows some errors due to RC